bundle/compliance/cis_gcp/rules/cis_4_7/rule.rego

```diff
@@ -0,0 +1,20 @@
+package compliance.cis_gcp.rules.cis_4_7
+
+import data.compliance.lib.common
+import data.compliance.policy.gcp.data_adapter
+
+default is_disk_encrypted_with_csek = false
+
+# Ensure VM Disks for Critical VMs Are Encrypted With Customer-Supplied Encryption Keys (CSEK)
+finding = result {
+	# filter
+	data_adapter.is_compute_disk
+
+	# set result
+	result := common.generate_result_without_expected(
+		common.calculate_result(is_disk_encrypted_with_csek),
+		{"Compute instance": data_adapter.resource},
+	)
+}
+
+is_disk_encrypted_with_csek := data_adapter.resource.data.diskEncryptionKey.sha256 != ""
```

Comment:
just for consistency with [other patterns](https://github.com/elastic/csp-security-policies/pull/286):
```suggestion
is_disk_encrypted_with_csek if {
    data_adapter.resource.data.diskEncryptionKey.sha256 != ""
} else = false
```